### ISSUE TITLE
Accessibility Elements Are Accessible Even Though Parent Has `accessibilityElementsHidden` Set To `true`

### ISSUE THREAD

**Issue by fruitcoder** [TextureGroup/Texture#694]
Since I'm fighting with this issue for several days now, I finally created a simple sample project to reproduce my problem. The issue is that `accessibilityElementsHidden` doesn't seem to be working in Texture. In the sample project you can see a simple hierarchy with two overlapping nodes containing a text node (which is accessible). When setting  `accessibilityElementsHidden` to `true` for the underlying node, the text node (which is a direct subnode) should not be accessible via UITests/VoiceOver etc. but with the AccessbilityInspector you can easily verify that it's still accessible. 

The weird thing is that the properties are correctly synced to the underlying UIViews, so 
`p (BOOL)[(UIView *)0x7fd573216310 accessibilityElementsHidden]` will be `YES` for the address of the node1's view.

The second tab shows the expected behavior in UIKit: The label1 is not accessible, since it's superview has  `accessibilityElementsHidden` set to `true`.

Texture Version: Texture (2.5.1)

[AccessibilityBug.zip](https://github.com/TextureGroup/Texture/files/1517860/AccessibilityBug.zip)

**Comment by fruitcoder** [comment]
Maybe I could fix the issue if you could point me in the right direction :)

**Comment by maicki** [comment]
@fruitcoder Sorry for the late response. Please look into `_ASDisplayViewAccessiblity.m`. Just looking over it briefly it seem we don't consider `accessibilityElementsHidden` while providing the accessibility elements to UIKit.

**Comment by fruitcoder** [comment]
Since the standard behavior of UIKit is to ignore those elements, shouldn't we filter the elements before handing them over to UIKit?

**Comment by fruitcoder** [comment]
I think we should return early from `CollectAccessibilityElementsForView` like this (line 219).  
![screen shot 2018-02-07 at 18 30 12](https://user-images.githubusercontent.com/5515849/35931458-fd0a5af0-0c34-11e8-9fa2-9c555f91166e.png)

**Comment by maicki** [comment]
@fruitcoder I just wanted to point you to the code :) Happy to review any PR about it.

**Comment by fruitcoder** [comment]
#795
